lib/class-wp-rest-url-details-controller.php

```diff
@@ -261,4 +372,205 @@ private function set_cache( $key, $data = '' ) {
 
 		return set_transient( $key, $data, $cache_expiration );
 	}
+
+	/**
+	 * Picks a random user agent string from a list of common defaults.
+	 * By default WordPress HTTP functions uses a semi-static string and
+	 * this maybe rejected by many websites.
+	 *
+	 * See: https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/class-http.php#L191.
+	 *
+	 * @return string the user agent string.
+	 */
+	private function get_random_user_agent() {
```

Comment:
>We can revert as the functionality can still be achieved via a filter if folks want to use my method in their own Plugin/code.

@getdave did you mean exposing this method for plugins/code? 

If yes, I'd suggest making this method static:

```suggestion
	public static function get_random_user_agent() {
```

Why? 
- Easier to invoke.
- And it's no longer directly associated with the specifics of an object; rather it's now a helper.

That said, a helper like this one seems out-of-place here in this class. When a random user agent is needed, this type of helper might be better served from the REST or HTTP API.

Reply:
No I meant you can [filter the args](https://github.com/WordPress/gutenberg/blob/411291d5b7fb26c93b6d777a3f478a97797d0eee/lib/class-wp-rest-url-details-controller.php#L187) in a Plugin or your own code. Agree we should remove the method from the `class`.